### PR TITLE
add sudo support of datahub-ingest's Dockerfile. [issue] https://gith…

### PR DESCRIPTION
**Pull request by hhofchina**
[sudo support of datahub-ingestion Dockerfile](https://github.com/linkedin/datahub/issues/4384\#issue-1166222918)

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by xiphl** [comment]
I do feel that putting sudo in a dockerfile is overkill, what are you installing after the image is built?

**Comment by hhofchina** [comment]
> 
In my pipeline, I want to add some software like golang, etc to the image to invoke datahub CLI directly. I have to integrate datahub CLI with other tools.  The datahub-ingestion project is too huge to port easily.

**Comment by hhofchina** [comment]
> I do feel that putting sudo in a dockerfile is overkill, what are you installing after the image is built?

In my application scenario, I am trying to provide integrated datahub metadata ingest capability service by calling datahub CLI directly, such as "datahub ingest -c config.yaml". In this way, there is no need to rely heavily on the datahub family bucket services.

File: build.gradle [hunk 1]
```diff
@@ -196,12 +196,12 @@ subprojects {
 
   tasks.withType(JavaCompile).configureEach {
     javaCompiler = javaToolchains.compilerFor {
-      languageVersion = JavaLanguageVersion.of(8)
+      languageVersion = JavaLanguageVersion.of(11)
```
Comment (anshbansal) [inline build.gradle:199]: Why is this change here? How is this related to adding `sudo`?

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#docker buildx build --platform linux/arm64 -t meshlake/datahub-ingestion:arm64 --push -f ./docker/datahub-ingestion/Dockerfile .
```
Comment (anshbansal) [inline build.sh:1]: Please remove this file. Why would the OSS have `meshlake/datahub-ingestion` in the source code? This should be in your local system
